packages/openapi3/src/types.ts

```diff
@@ -6,12 +6,41 @@ export type Extensions = {
 
 export interface OpenAPI3Document extends Extensions {
   openapi: "3.0.0";
+
+  /**
+   * Provides metadata about the API. The metadata can be used by the clients if needed.
+   */
   info: OpenAPI3Info;
-  paths: any;
-  servers?: OpenAPI3Server[];
+
+  /** Additional external documentation. */
   externalDocs?: OpenAPI3ExternalDocs;
+
+  /** The available paths and operations for the API */
+  paths: Record<string, OpenAPI3PathItem>;
+
+  /**
+   * Additional paths and operations that cannot be used in `paths` as the url is not unique.
+   * This can be used to provide operation overload using a query param.
+   * @example "/foo?type=abc" returning FooA and "/foo?type=xyz" returning FooB are not allowed in `paths` as there is query params.
+   */
+  "x-ms-paths"?: Record<string, OpenAPI3PathItem>;
```

Comment:
Does the OpenAPI 3 emitter support `x-ms-paths`?  I think we intended to keep MS-specific extensions out of this emitter